### PR TITLE
[SOAR-17270,SOAR-17272 ] Mimecast | Snyk fix | Improve task connection test logic

### PR DESCRIPTION
**Pull request by cmcnally-r7**
### Description

Describe the proposed changes:

https://rapid7.atlassian.net/browse/SOAR-17270
  - Update setuptools to `70.0.0`
  - Update version to `5.3.15`

https://rapid7.atlassian.net/browse/SOAR-17272
 After new changes to the SDK, we now support passing back a message to the SDK as part of the task connection test, this will allow for the use of the custom message to be shown in the UI for a task connection test, rather than have the end user see all of the available logs 

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Testing

This has been tested locally using docker and postman, using the mew version of the SDK

on a successful test, task route 
image

on a successful test, connection test route 
image

on a failure we will now try and return the data section of the error raised to the end user rather than the fill log, in this case Mimecast will use the case and exception fields to build the data 
 
image

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/mimecast/plugin.spec.yaml [hunk 1]
```diff
@@ -17,15 +17,15 @@ links:
  - "[Mimecast](http://mimecast.com)"
 references:
  - "[Mimecast API](https://www.mimecast.com/developer/documentation)"
-version: 5.3.14
+version: 5.3.15
 connection_version: 5
 supported_versions: ["Mimecast API 2024-05-09"]
```
Comment (ablakley-r7) [inline plugins/mimecast/plugin.spec.yaml:22]: Can we update this to last test date?

File: plugins/mimecast/komand_mimecast/connection/connection.py [hunk 2]
```diff
@@ -43,9 +43,24 @@ def test_task(self):
...
         except ApiClientException as error:
-            self.logger.info("The connection test to Mimecast has failed")
+
+            return_message = ""
+
+            failed_message = "The connection test to Mimecast has failed"
+            self.logger.info(failed_message)
+            return_message += f"{failed_message}\n"
+
+            cause_message = f"This failure was caused by: {error.cause}"
```
Comment (ablakley-r7) [inline plugins/mimecast/komand_mimecast/connection/connection.py:57]: These look fine to me. Do we have examples of the cause and assistance outside of the standardised error messages? I know sometimes the assistance messages don't provide a direct solution only guidance.

It's less explicit but we could use `cause:` and `assistance` if there's more variability in the values.

Comment (rbowden-r7) [reply plugins/mimecast/komand_mimecast/connection/connection.py:57]: The main list of the cause and assistance come form here - https://github.com/rapid7/insightconnect-plugins/blob/f2ccfe5052e5f77c9ba09f46428822e05bbbde3b/plugins/mimecast/komand_mimecast/util/api.py#L316 

This is still open to debate if there are better solutions this was just me trying to reuse as much of what was already there to get something of the ground

File: plugins/mimecast/plugin.spec.yaml (hunk 1 above)
Comment (rbowden-r7) [reply plugins/mimecast/plugin.spec.yaml:22]: This has been updated

**Review by igorski-r7 — APPROVED** [review]

File: plugins/mimecast/komand_mimecast/connection/connection.py [hunk 3]
```diff
@@ -43,9 +43,24 @@ def test_task(self):
...
+            self.logger.info(cause_message)
+            return_message += f"{cause_message}\n"
+
+            assistance_message = f"This failure is fixed by: {error.assistance}"
```
Comment (joneill-r7) [inline plugins/mimecast/komand_mimecast/connection/connection.py:61]: I think most of the assistances don't fit in with this prefix "This failure is fixed by:". Could just append the assistance to the end? At the minute in the PR reads as "This failed is fixed by: Please ensure that correct APP ID is supplied"

File: plugins/mimecast/komand_mimecast/connection/connection.py (hunk 2 above)
Comment (joneill-r7) [reply plugins/mimecast/komand_mimecast/connection/connection.py:57]: Another minor, should the cause be in nested quotes? `"This failure was caused by 'Incorrect App ID supplied'"` ?

File: plugins/mimecast/komand_mimecast/connection/connection.py (hunk 3 above)
Comment (rbowden-r7) [reply plugins/mimecast/komand_mimecast/connection/connection.py]: this has been updated to remove the extra wording so it should now read better

File: plugins/mimecast/komand_mimecast/connection/connection.py (hunk 2 above)
Comment (rbowden-r7) [reply plugins/mimecast/komand_mimecast/connection/connection.py:57]: This has been updated to include single `''` around the cause field

**Review by joneill-r7 — APPROVED** [review]

**Pull request merged**